client/sites/settings/controller.tsx

```diff
@@ -43,6 +42,14 @@ export function siteSettings( context: PageJSContext, next: () => void ) {
 }
 
 export function administrationSettings( context: PageJSContext, next: () => void ) {
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+	const siteSlug = getSelectedSiteSlug( state );
+
+	if ( isSiteWpcomStaging( state, siteId ) ) {
+		return page.redirect( `/overview/${ siteSlug }` );
+	}
```

Comment:
Ideally this should be redirecting to `/sites/settings/administration` and the site selector that gets shown there should be filtering out wpcomstaging sites from the list of sites to choose from.

Basically I want to avoid the situation a user follows a link to `/sites/settings/administration` in the help docs, selects their staging site, and then ends up somewhere they didn't ask for. I'd rather prevent the scenario in the first place.

If that level of filtering isn't possible we should not gate the settings page at all and just display an empty state explaining why the options are not available.

It's also a signal to me that these settings pages are too broken down and probably need consolidating, e.g. administration, site, agency are probably best as some kind of general settings page.

@lucasmendes-design in case you have thoughts about this user interaction.

Reply:
Maybe we should just show it? I don't know why it's gated from staging sites. It looks like there's request to open it up to staging sites, too:

- https://github.com/Automattic/wp-calypso/issues/90368

In any case, without (any) redirection, the current behavior is weird. Please check the test instructions. If we're on Agency/Administration tab, and then select another site from the site list sidebar, the following approaches feel broken:

- Showing site selector (why do I get shown this when all I did is select another site in the sites list)
- Showing a blank page (current behavior) without active sidebar item tab

Maybe the best approach is to redirect to `/overview` and showing a snackbar saying that "Agency/admin setting is not supported here..."

 > are probably best as some kind of general settings page.

I'm afraid people will confuse it with Settings -> General from core wp-admin.

Reply:
I'll try to timebox and experiment with something/

Reply:
I changed my approach; it's basically similar to your previous idea, but with this it can show the sidebar item to avoid confusion. Please check the new changes and test instructions. 😄

Reply:
I think this works ok, the collocation in controller is an improvement, there's still going to be duplicate logic in the inner form but I'm ok with that.